ipq95xx: Add support for IPQ9574 RDP433

quic-varada opened the pull request:
Hi,

This pull request is for adding support to the QCOM IPQ9574 SoC.
Kindly review and provide feedback.

SoC	: QCOM IPQ9574
RAM	: 2GB DDR4
Flash	: eMMC 8GB
WiFi	: 1 x 2.4GHz
	  1 x 5GHz
	  1 x 6GHz

Thanks
Varada

xback commented:
@quic-varada
_ipq95xx: Enable RPM based regultors_ --> regulators

mrkiko commented:
Mark to follow.

quic-varada commented:
> @quic-varada _ipq95xx: Enable RPM based regultors_ --> regulators
> 
> Also, get rid of the merge commit:
> 
> * pull + rebase
> * force-push

Thanks for the feedback. Have updated and pushed once again.
Please take a look.

981213 commented:
Hi! This PR looks good to me so far.
But as we discussed before on mail, a serial-only target isn't really usable for OpenWrt. It should at least get working ethernet before it can be merged.
Don't be hesitant to keep working on it and push further changes here for reviewing :)

rmandrad commented:
> Hi! This PR looks good to me so far. But as we discussed before on mail, a serial-only target isn't really usable for OpenWrt. It should at least get working ethernet before it can be merged. Don't be hesitant to keep working on it and push further changes here for reviewing :)

what is the device here ? a target is fine but where is this has been tested and proven?

ynezz reviewed (changes requested):
@quic-varada thanks, but is there any reason, why you didn't already incorporated the review feedback provided [on the patch](https://patchwork.ozlabs.org/project/openwrt/patch/20231207095923.3913903-1-quic_varada@quicinc.com/) you've sent to the mailing list?

I see namely following most glaring points:

1. [Kernel patches - please sort the patches by prefix such as](https://patchwork.ozlabs.org/project/openwrt/patch/20231207095923.3913903-1-quic_varada@quicinc.com/#3229858):
    * 0xx are backports (Kernel version from which they are backported must be marked as well)
       * `ls -al target/linux/qualcommax/patches-6.1/*v6.*` for prior examples  
    * 1xx are pending
    * 9xx are usually hacks/stuff that currently cannot be upstreamed

2. [target and subtarget name](https://patchwork.ozlabs.org/project/openwrt/patch/20231207095923.3913903-1-quic_varada@quicinc.com/#3230401)
   * IIUC then the current preference is to use `qualcommax/ipq95xx`, we can always rename/split it later, so no worries, its not something set into the stone.

In other words, those are mandatory.

ynezz commented:
> a serial-only target isn't really usable for OpenWrt
> It should at least get working ethernet before it can be merged.

I would say, that its not strictly mandatory, see such prior examples 29c6ddd99ee and 3c1f6e358d. I for example find this smaller diffs much easier for handling/reasoning about for both sides (contributor/reviewer). Looking at the diff, its mostly upstream backports right now.

And there was a promise made: [Yes, there is plan to post those functionalities in the coming weeks.](https://patchwork.ozlabs.org/project/openwrt/patch/20231207095923.3913903-1-quic_varada@quicinc.com/#3230401) 

> what is the device here ?

IIUC then its a `ap-al02.1-c4` board targeting developers, so QCA's development kit.

> a target is fine but where is this has been tested and proven?

Yep, I'll test this.

quic-varada commented on target/linux/ipq95xx/base-files/etc/inittab:
```diff
@@ -0,0 +1,18 @@
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
```
My mistake.
Will make the corrections and post a new one.

Thanks

huangya90 commented:
Hi, does ipq95xx still use "switch_mac_mode" to configure interface mode? If yes, why I can not see them in the dts?

Yewenzhou1994 commented:
您好，请问可以适配一下 TP-Link TL-7TR13090 吗？这个机器能使用320Mhz

AEGEGE commented:
Such a good project hasn't been updated for a long time, what's going on?

tr0llex commented:
@ynezz @quic-varada Hello there. The community of Xiaomi Router 7000 (Xiaomi Router BE7000) owners, represented by the 4PDA forum members, is really looking forward to adding support for this SoC. Can we help in any way to get this Pull Request moving forward?

zinky81 commented:
Also new routers are appearing with IPQ9574. The latest one is Xiaomi 10G (BE10000). This should be very interesting router as it is successor of very popular Xiaomi AX9000, but with support of WIFI-7. This pull request should have a much more attention as device list is growing rapidly.

zmashine commented:
**[@ynezz](https://github.com/ynezz)** [@quic-varada](https://github.com/quic-varada)
 
You are doing a very good thing. People crave more freedom and hope for you very much. Bring the case to the end. Thank you very much.

zmashine commented:
@ynezz @quic-varada
 
You are doing a very good thing. People crave more freedom and hope for you very much. Bring the case to the end. Thank you very much.

quic-rjangir committed:
ipq95xx: Add basic IPQ9574 SoC support

 SoC	: QCOM IPQ9574
 RAM	: 2GB DDR4
 Flash	: eMMC 8GB, NAND 128MB
 WiFi	: 1 x 2.4GHz
	  1 x 5GHz
	  1 x 6GHz

Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>

quic-rjangir committed:
ipq95xx: Enable spi-qpic driver for QCOM SPI NAND flash Interface

This driver implements support for the SPI-NAND mode of QCOM NAND Flash
Interface as a SPI-MEM controller with pipelined ECC capability.

Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>

frogost commented:
I have a Xiaomi Router 7000 (Xiaomi Router BE7000) and I am really looking forward to the implementation of this project

quic-varada commented:
Hi All,

the above comments are addressed in updated PR. Kindly help to review.

Thanks,
Varada

981213 commented:
Hello! It's been half a year since the promised ethernet support. Any updates on that?

BotMolotovich commented:
Hi! watching this pr for monthes, will so much appreciate any updates here!

ArtMvs commented:
I have been following since February and every month I am disappointed, when will you stop our suffering? Why does your direct competitor mtk provide all the sources and moreover helps to develop openwrt?

Xiaomi be7000

isikerkan commented:
> I have been following since February and every month I am disappointed, when will you stop our suffering? Why does your direct competitor mtk provide all the sources and moreover helps to develop openwrt?
> 
> Xiaomi be7000

Problem here is not the Devs @ArtMvs . Qualcom Employees coded the Drivers and submitted the fixes, but no reviewer checked them out yet. @ynezz Could you check the updated PR out?

We are eagerly waiting to flash the BE10000 to OpenWRT to have a affordable 10 Gbit/s SPF Router with a customizable OS.

981213 commented:
@isikerkan Please note that this PR does not bring any networking support, meaning it's far from being usable on a real router.
The currently submitted source code has been reviewed and acked a long time ago, since it contains mostly backported patches. I just don't feel like merging such a target where nothing other than the serial console works.

981213 commented on target/linux/qualcommax/patches-6.6/0409-Squashfs-Accept-mtd-names-for-mounting.patch:
```diff
@@ -0,0 +1,79 @@
+@@ -480,8 +484,44 @@ failed_mount:
+ 	return err;
+ }
+ 
```
Drop this patch. This hack shouldn't be needed on mainline OpenWrt.

981213 commented on target/linux/qualcommax/patches-6.6/0411-fix-build-errors.patch:
```diff
@@ -0,0 +1,58 @@
+ 	struct mtd_blktrans_ops *tr;
+ 
+-	if (mtd->type == MTD_ABSENT || mtd->type == MTD_UBIVOLUME)
++	if (mtd->type == MTD_ABSENT)
```
Drop this change to the mtd core unless you have a really good reason to do so. If it's absolutely needed, please split it into a separated patch and explain why in the mbox header of the new patch.

AEGEGE commented:
It seems that QCOM IPQ9574 will have a long way to go to successfully run openwrt. I am disappointed.

RightDecision2 commented:
> It seems that QCOM IPQ9574 will have a long way to go to successfully run openwrt. I am disappointed.

Why you cant take needed files/drivers from Xiaomi BE7000?
He ships with OpenWRT 18.06_Snapshot and all works well with QCOM

zinky81 commented:
This topic about support on IPQ9574 is somehow strange, because there is already functional QWRT firmware for Xiaomi BE10000 and BE7000 which is possible to purchase somewhere on Chinese and Russian forums. I'm also running this firmware and I can confirm it's running on kernel version 5.4.164. I have also tried to compile my own OpenWRT because there is already listed IPQ9574 support but it was unsuccessful due to limitations in Linux kernel for BE (Wifi-7). So for sure it can work, but the question is why is not open for OPENWRT. Thanks!

waitingsnow commented:
> This topic about support on IPQ9574 is somehow strange, because there is already functional QWRT firmware for Xiaomi BE10000 and BE7000 which is possible to purchase somewhere on Chinese and Russian forums. I'm also running this firmware and I can confirm it's running on kernel version 5.4.164. I have also tried to compile my own OpenWRT because there is already listed IPQ9574 support but it was unsuccessful due to limitations in Linux kernel for BE (Wifi-7). So for sure it can work, but the question is why is not open for OPENWRT. Thanks!

How can I get the firmware you mentioned? I need it very much, thank you

AEGEGE commented:
> This topic about support on IPQ9574 is somehow strange, because there is already functional QWRT firmware for Xiaomi BE10000 and BE7000 which is possible to purchase somewhere on Chinese and Russian forums. I'm also running this firmware and I can confirm it's running on kernel version 5.4.164. I have also tried to compile my own OpenWRT because there is already listed IPQ9574 support but it was unsuccessful due to limitations in Linux kernel for BE (Wifi-7). So for sure it can work, but the question is why is not open for OPENWRT. Thanks!

Xiaomi BE10000 hardware is not fully functional. The TP-LINK TL-7TR-13090 BE13000 I have has basically full performance. I can't find any information about shell and openwrt. I don't know when I can get it.

RightDecision2 commented:
[Xiaomi 7000.zip](https://github.com/user-attachments/files/18160232/Xiaomi.7000.zip)
Please review these files from Xiaomi BE7000.
IPQ9574.bin -> /tmp/IPQ9574/caldata.bin
qcn9224.bin -> /tmp/qcn9224/caldata_3.bin

ndoo commented:
More IPQ95xx targets that have retail availability and would greatly benefit from this PR as a starting point:

- [Alta Labs Route10](https://www.alta.inc/route10)
- Most TP-Link 802.11be Products (Deco BE95, Deco BE85, EB810v, HB810)

+1 to get this going, meanwhile will experiment locally with QSDK tree.

A-SOM commented:
any updates? What can I do to help speed up implementation?

dima424658 commented:
> How can I get the firmware you mentioned? I need it very much, thank you

@waitingsnow , here is the firmware for be7000 - https://cdn.cnbj1.fds.api.mi-img.com/xiaoqiang/rom/rc06/miwifi_rc06_firmware_93757_1.1.16.bin
it can be downloaded from - https://www1.miwifi.com/miwifi_download.html

A-SOM commented:
> I have not had time to give it a go, but in the meantime, I have requested and received the GPL archive for Alta Labs AP6:
> 
> https://dl.alta.inc/gpl/gpl-2.2f-zkdp29d9.tar.gz

Hello. You can check gpl archive too here for MBE70:
https://support.linksys.com/kb/article/316-en/

https://downloads.linksys.com/support/assets/gpl/MBE70WRT_v1.1.0.01.tgz

PerezChilli commented:
Mee too join and up this thread. I'm hope support will be reached cause IPQ95XX becomes more and more popular platform...

ynezz commented:
Thanks, but it was superseded by https://github.com/openwrt/openwrt/pull/17725

peterwillcn commented:
any news?
